lib/strava/web/response.rb

```diff
@@ -0,0 +1,31 @@
+module Strava
+  module Web
+    module Response
+      def extract_response!(response)
+        extract_ratelimit!(response)
+      end
+
+      def extract_ratelimit!(response)
+        return response unless response.body.present?
+
+        case response.body.class.name
+        when 'Array'
+          response.body.map! do |elem|
+            merge_ratelimit!(elem, response)
```

Comment:
It's possible that `response.body` is frozen, so you cannot modify it in-place. Assign ratelimit instance to the response object.